Turn preloading on by default for cached requests (#1489)
Shopify/hydrogen: docs/hooks/global/fetchsync.md

The `requestInit` object augments the [`init` properties available in the Web Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request) to include the following additional properties:

| Key                   | Required | Description                                                                                                                                                                                                                                    |
| --------------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `cache`               | No       | The [caching strategy](https://shopify.dev/custom-storefronts/hydrogen/framework/cache#caching-strategies) to help you determine which cache control header to set.                                                                            |

> | `preload`             | No       | Whether to [preload the request](https://shopify.dev/custom-storefronts/hydrogen/framework/preloaded-queries). It defaults to true only when the `CachingStrategy` is not `NoStore`. Specify `false` to disable or use `'*'` to preload the query for all requests. |

Reviewer: ```suggestion
| `preload`             | No       | Whether to [preload the request](https://shopify.dev/custom-storefronts/hydrogen/framework/preloaded-queries). It defaults to `true` only when the `CachingStrategy` value isn't `NoStore`. Specify `false` to disable or use `'*'` to preload the query for all requests. |
```
